### DISCUSSION TITLE
Flavors and Paparazzi

### DISCUSSION THREAD

**Discussion by eschoenawa** [cashapp/paparazzi#540] [Q&A] [answered]
Hello everyone,
I have a question regarding app build flavors. The app I'm working on has multiple build flavors to serve the same app in very different styling. This is realized by overwriting the themes used within all UI in each build flavor with different styles. How can I specify the build flavor for running paparazzi tests? I can't specify the wanted theme directly with the `theme` parameter, as the theme has the same name as the default one, in order to overwrite it.

I also can't supply the flavor through the gradle task, as only our app module has the flavors specified, and that module can't run Paparazzi (due to not being a lib module). Our lib modules don't have the flavors in the gradle tasks.

**Comment by erawhctim** [comment, marked as answer]
Paparazzi supports application modules now (in v1.3.1+), so you could try and run the paparazzi gradle plugin directly on your app module. 

If you can't upgrade to the latest Paparazzi version (e.g. if you're on an older version of Gradle), you could try declaring your app module as an Android library module (call it `:app-library`), and then creating a new, nearly-empty app module that has a dependency on your `:app-library` module instead. It's a cheap way to get around Paparazzi's app module limitation on versions earlier than 1.3.1.
